### PR TITLE
[GH-386] add option anonymous creator

### PR DESCRIPTION
**Pull request by annaos**
- Add new CLI option `--anonymous-creator `
- Add new option to dialog
- Add tests

Issue: #386

### CODE REVIEW

**Comment by hanzei** [comment]
@annaos is your branch up to date with `master`?

**Comment by annaos** [comment]
> @annaos is your branch up to date with `master`?

yes, it is up to date

File: server/poll/transform.go [hunk 1]
```diff
@@ -167,6 +171,9 @@ func (p *Poll) makeAdditionalText(bundle *utils.Bundle, numberOfVotes, numberOfV
 	if p.Settings.Anonymous {
 		settingsText = append(settingsText, "anonymous")
 	}
+	if p.Settings.AnonymousCreator {
+		settingsText = append(settingsText, "anonymous-creator")
+	}
```
Comment (hanzei) [inline server/poll/transform.go:176]: Unrelated to the PR: Maybe `poll.Settings` needs a `String()` method.

**Review by hanzei — APPROVED** [review]
Great work, LGTM :rocket:

**Comment by hanzei** [comment]
The CI is unrelated and will be fixed on the next commit.

File: server/poll/transform.go (hunk 1 above)
Comment (annaos) [reply server/poll/transform.go]: I did it, but I am unsure, if it is good idea and if it need (so much) tests. At the moment I can found only one place to use this function. It is ok for me to revert last commit :)

**Review by kaakaa — CHANGES_REQUESTED** [review]
Thanks @annaos !

This work looks great, but a creator name still remain in a card, and `Created by` section in a card should also be hidden with `anonymous-creator` setting. (Card can be displayed only if a poll is create with `--progress` setting)

スクリーンショット 2024-03-31 22 28 16

Could you fix this? Related code would be here. https://github.com/matterpoll/matterpoll/blob/master/server/poll/transform.go#L260

**Review by kaakaa — APPROVED** [review]
Thanks 👍

**Pull request merged**